x/lockup/keeper/lock_test.go

```diff
@@ -96,143 +65,156 @@ func (suite *KeeperTestSuite) TestGetPeriodLocks() {
-	suite.Require().Len(locks, 0)
-}
-
 func (suite *KeeperTestSuite) TestUnlock() {
-	// test for coin unlocking
 	suite.SetupTest()
-	now := suite.Ctx.BlockTime()
+	initialLockCoins := sdk.Coins{sdk.NewInt64Coin("stake", 10)}
+
+	testCases := []struct {
+		name                          string
+		unlockingCoins                sdk.Coins
+		expectedBeginUnlockPass       bool
+		passedTime                    time.Duration
+		expectedUnlockMaturedLockPass bool
+		balanceAfterUnlock            sdk.Coins
+	}{
+		{
+			name:                          "normal unlocking case",
+			unlockingCoins:                initialLockCoins,
+			expectedBeginUnlockPass:       true,
+			passedTime:                    time.Second,
+			expectedUnlockMaturedLockPass: true,
+			balanceAfterUnlock:            initialLockCoins,
+		},
+		{
+			name:                          "begin unlocking with nil as parameter",
+			unlockingCoins:                nil,
+			expectedBeginUnlockPass:       true,
+			passedTime:                    time.Second,
+			expectedUnlockMaturedLockPass: true,
+			balanceAfterUnlock:            initialLockCoins,
+		},
+		{
+			name:                          "unlocking coins exceed what's in lock",
+			unlockingCoins:                sdk.Coins{sdk.NewInt64Coin("stake", 20)},
+			expectedBeginUnlockPass:       false,
+			passedTime:                    time.Second,
+			expectedUnlockMaturedLockPass: false,
+			balanceAfterUnlock:            sdk.Coins{},
+		},
+		{
+			name:                          "unlocking unknown tokens",
+			unlockingCoins:                sdk.Coins{sdk.NewInt64Coin("unknown", 10)},
+			expectedBeginUnlockPass:       false,
+			passedTime:                    time.Second,
+			expectedUnlockMaturedLockPass: false,
+			balanceAfterUnlock:            sdk.Coins{},
+		},
+		{
+			name:                          "partial unlocking",
+			unlockingCoins:                sdk.Coins{sdk.NewInt64Coin("stake", 5)},
+			expectedBeginUnlockPass:       true,
+			passedTime:                    time.Second,
+			expectedUnlockMaturedLockPass: true,
+			balanceAfterUnlock:            sdk.Coins{sdk.NewInt64Coin("stake", 5)},
+		},
+		{
+			name:                          "partial unlocking unknown tokens",
+			unlockingCoins:                sdk.Coins{sdk.NewInt64Coin("unknown", 5)},
+			expectedBeginUnlockPass:       false,
+			passedTime:                    time.Second,
+			expectedUnlockMaturedLockPass: false,
+			balanceAfterUnlock:            sdk.Coins{},
+		},
+		{
+			name:                          "unlocking should not finish yet",
+			unlockingCoins:                initialLockCoins,
+			expectedBeginUnlockPass:       true,
+			passedTime:                    time.Millisecond,
+			expectedUnlockMaturedLockPass: false,
+			balanceAfterUnlock:            sdk.Coins{},
+		},
+	}
 
-	addr1 := sdk.AccAddress([]byte("addr1---------------"))
-	coins := sdk.Coins{sdk.NewInt64Coin("stake", 10)}
+	for _, tc := range testCases {
+		suite.SetupTest()
```

Comment:
I see that we do `suite.App.LockupKeeper`, `suite.App.BankKeeper` and `suite.Ctx` a lot. What do you thnk about initializing variables like:
```
lockupKeeper := suite.App.LockupKeeper
bankKeeper := suite.App.BankKeeper
ctx := suite.Ctx
```
This might help reduce verbosity, making the logic easier to read